### ISSUE TITLE
rfc: add DetailsList to/in FAST

### ISSUE THREAD

**Issue by hkusulja** [microsoft/fast#5442]
Hello,
we are comparing Microsoft Fluent UI React - web components - such as DetailsList - https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist which has lot of options, features etc.
To a Microsoft FAST Fluent UI using Blazor WebAssembly (clientside) .NET 6 .
Also, I am referencing Github issue from FAST-Blazor repo - https://github.com/microsoft/fast-blazor/issues/47
FAST-Blazor teams says to open here Github issue.

Can you please align expectations here, when and if will Microsoft FAST have a functionality on grids / DetailsList like sorting, filtering, marquee selection, justified columns, modal selection, compact mode, drag-n-drop, grouping etc.?

Current FAST and Blazor component is very very basic, with almost none mentioned features.

**Comment by scomea** [comment]
I don't believe this is on the books yet, but I would definitely favor an approach where we have complex components that are derived from the base grid, like a DetailsList could be, rather than try to make the base grid overly feature heavy.  We already have a calendar component that is based on data-grid.  https://github.com/microsoft/fast/blob/746ed487e8eb1ab2ede0ecdc09a6a11afc0baaba/packages/web-components/fast-foundation/src/calendar/calendar.template.ts#L128

**Comment by hkusulja** [comment]
Ok, it does not have to be in grid component if you want it lightweight, it can be separate componentent called DetailsList, but still, can we hope this (almost all requested features, sorting, filterin, modalselect etc.) will be resolved here soon, or we have to find out other solutions?

**Comment by EisenbergEffect** [comment]
@chrisdholt You are probably best to respond to this. I don't even know if DetailsList is part of Fluent anymore, for example.

**Comment by chrisdholt** [comment]
> @chrisdholt You are probably best to respond to this. I don't even know if DetailsList is part of Fluent anymore, for example.

I see this, I'm going to do some looking to see what the timeline/trajectory is. What I can say is that if it *is* part of Fluent it's likely an implementation of Grid. If it's something else, we can certainly look at adding that new interaction model but my understanding is that this would be an implementation of the FAST Grid in the Fluent visual language, likely with some additional ergonomics to address the common scenarios you're mentioning. Depending on what I find we can look to move this over to the Fluent repo.

**Comment by hkusulja** [comment]
But we need this such a control which existis in Fluent UI React, to the Blazor FAST ....

**Comment by EisenbergEffect** [comment]
@hkusulja We hear you and would certainly like to be able to offer this control to you as part of this library. That said, at this time, we are unable to make a commitment on this particular work item for a variety of reasons. I'm happy to keep this issue open so that we can keep you updated and provide you with more information as the situation changes.

**Comment by hkusulja** [comment]
@EisenbergEffect fair enough, thank you for the answer.

**Comment by hkusulja** [comment]
@EisenbergEffect told in latest that this will be kept open, but now is auto-closed, so please reopen

**Comment by chrisdholt** [comment]
> @EisenbergEffect told in latest that this will be kept open, but now is auto-closed, so please reopen

I'm going to move this over to the Fluent web components as this is an implementation of several foundational components and not a Foundational component. Happy to track over there - I'll make the move accordingly.
